core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+      expect(infoWrapper).toBeInTheDocument();
+    });
+
+    it('does not render info tooltip when disabled', () => {
+      const { container } = render(<Input name="test" info="This is helpful information" disabled={true} />);
+
+      const infoWrapper = container.querySelector('[tabIndex="0"]');
+      expect(infoWrapper).not.toBeInTheDocument();
+    });
+  });
+
+  describe('Focus management', () => {
+    it('focuses input programmatically when needed', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+
+      input.focus = jest.fn();
+      input.focus();
+
+      expect(input.focus).toHaveBeenCalled();
+      expect(input).toBeInTheDocument();
+    });
+
+    it('sets tabIndex to -1 when readOnly is true', () => {
+      const { getByTestId } = render(<Input name="test" readOnly={true} />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      expect(input.tabIndex).toBe(-1);
+    });
+
+    it('does not set tabIndex when readOnly is false', () => {
+      const { getByTestId } = render(<Input name="test" readOnly={false} />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      expect(input.tabIndex).not.toBe(-1);
+    });
+  });
+
+  describe('State management', () => {
+    it('updates isInputBlank state when wrapper loses focus', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+
+      fireEvent.change(input, { target: { value: 'test' } });
+
+      fireEvent.blur(wrapper);
+
+      expect(wrapper).toBeInTheDocument();
+    });
+  });
+
+  describe('Error handling', () => {
+    it('handles cursor positioning error gracefully', () => {
+      const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+
+      const TestComponent = () => {
+        const [inputType, setInputType] = React.useState<'text' | 'number'>('text');
+
+        React.useEffect(() => {
+          setInputType('number');
+        }, []);
+
+        return <Input name="test" type={inputType} value="123" />;
+      };
+
+      render(<TestComponent />);
+
+      consoleErrorSpy.mockRestore();
+    });
+  });
+
+  describe('Different input types', () => {
+    const inputTypes: Array<'text' | 'password' | 'number' | 'email' | 'tel' | 'url'> = [
+      'text',
+      'password',
+      'number',
+      'email',
+      'tel',
+      'url',
+    ];
+
+    inputTypes.forEach((type) => {
+      it(`renders correctly with type="${type}"`, () => {
+        const { getByTestId } = render(<Input name="test" type={type} value="test" />);
+
+        const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+        expect(input.type).toBe(type);
+      });
+    });
+  });
+
+  describe('Size variants', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.